packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-timeline/atomic-insight-user-actions-timeline.tsx

```diff
@@ -0,0 +1,222 @@
+              data-testid={testId}
+            ></atomic-insight-user-actions-session>
+          </div>,
+          renderSeparator ? renderSeparator() : null,
+        ])}
+      </div>
+    );
+  }
+
+  private renderToggleFollowingSessionsButton() {
+    return (
+      <div class="flex justify-center p-2">
+        <button
+          onClick={this.toggleFollowingSessions.bind(this)}
+          class="btn-text-primary flex items-center p-2"
+          data-testid={
+            this.followingSessionsAreVisible
+              ? 'hide-following-sessions'
+              : 'show-following-sessions'
+          }
+        >
+          <atomic-icon
+            class="mr-1.5 h-3 w-3"
+            icon={
+              this.followingSessionsAreVisible ? ArrowDownIcon : ArrowUpIcon
+            }
+          ></atomic-icon>
+          {this.followingSessionsAreVisible
+            ? this.bindings.i18n.t('hide-following-sessions')
+            : this.bindings.i18n.t('show-following-sessions')}
+        </button>
+      </div>
+    );
+  }
+
+  private renderTogglePrecedingSessionsButton() {
+    return (
+      <div class="flex justify-center p-2">
+        <button
+          onClick={this.togglePrecedingSessions.bind(this)}
+          class="btn-text-primary flex items-center p-2"
+          data-testid={
+            this.precedingSessionsAreVisible
+              ? 'hide-preceding-sessions'
+              : 'show-preceding-sessions'
+          }
+        >
+          <atomic-icon
+            class="mr-1.5 h-3 w-3"
+            icon={
+              this.precedingSessionsAreVisible ? ArrowUpIcon : ArrowDownIcon
+            }
+          ></atomic-icon>
+          {this.precedingSessionsAreVisible
+            ? this.bindings.i18n.t('hide-preceding-sessions')
+            : this.bindings.i18n.t('show-preceding-sessions')}
+        </button>
+      </div>
+    );
+  }
+
+  private renderFollowingSessionsSection() {
+    if (!this.userActionsState.timeline?.followingSessions?.length) {
+      return null;
+    }
+    return [
+      this.renderToggleFollowingSessionsButton(),
+      <div class="separator mx-1 rounded"></div>,
+      this.followingSessionsAreVisible
+        ? this.renderSessions(
+            this.userActionsState.timeline?.followingSessions,
+            () => <div class="separator mx-1 mt-4 rounded"></div>,
+            'following-session'
+          )
+        : null,
+    ];
+  }
+
+  private renderPrecedingSessionsSection() {
+    if (!this.userActionsState.timeline?.precedingSessions?.length) {
+      return null;
+    }
+    return [
+      <div class="separator mx-1 mt-4 rounded"></div>,
+      this.precedingSessionsAreVisible
+        ? this.renderSessions(
+            this.userActionsState.timeline?.precedingSessions,
+            () => <div class="separator mx-1 mt-4 rounded"></div>,
+            'preceding-session'
+          )
+        : null,
+      this.renderTogglePrecedingSessionsButton(),
+    ];
+  }
```

Comment:
```suggestion
  private renderPrecedingSessionsSection() {
    if (!this.userActionsState.timeline?.precedingSessions?.length) {
      return null;
    }
    return [
      <div class="separator mx-1 mt-4 rounded"></div>,
      this.precedingSessionsAreVisible
        ? this.renderSessions(
            this.userActionsState.timeline?.precedingSessions,
            () => <div class="separator mx-1 mt-4 rounded"></div>,
            'preceding-session'
          )
        : null,
      this.renderTogglePrecedingSessionsButton(),
    ];
  }

  private renderFollowingSessionsSection() {
    if (!this.userActionsState.timeline?.followingSessions?.length) {
      return null;
    }
    return [
      this.renderToggleFollowingSessionsButton(),
      <div class="separator mx-1 rounded"></div>,
      this.followingSessionsAreVisible
        ? this.renderSessions(
            this.userActionsState.timeline?.followingSessions,
            () => <div class="separator mx-1 mt-4 rounded"></div>,
            'following-session'
          )
        : null,
    ];
  }
```

very minor but it makes more sense to have renderPrecedingSessionsSection first

Reply:
I don't see any added value in switching the order of the functions.
even if it mattered, the following session are displayed first in the UI.